odoo/osv/expression.py

```diff
@@ -923,7 +925,8 @@ def parent_of_domain(left, ids, left_model, parent=None, prefix=''):
                     records = records[parent_name] - records.browse(parent_ids)
                 domain = [('id', 'in', list(parent_ids))]
             if prefix:
-                return [(left, 'in', left_model_sudo._search(domain))]
+                rhs = list(parent_ids) or left_model_sudo._search(domain)
```

Comment:
You always have the `parent_ids`, then you can always return `[(left, 'in', list(parent_ids)]`.